### PR TITLE
Performance test

### PR DESCRIPTION
**Pull request by christopherlagerhult**

### CODE REVIEW

File: source/base/Profiler.ooc [hunk 1]
```diff
@@ -50,7 +52,7 @@ Profiler: class {
 		This _profilers free()
 		This _profilers = null
 	}
-	benchmark: static func (task: Func, timeoutMilliseconds := 1000.0) -> Double {
+	benchmark: static func ~closure (task: Func, timeoutMilliseconds := 1000.0) -> Double {
```
Comment (emilwestergren) [inline source/base/Profiler.ooc]: Yes, TimeSpan I like

Comment (christopherlagerhult) [inline source/base/Profiler.ooc]: Sure

Comment (christopherlagerhult) [inline source/base/Profiler.ooc]: #1961

**Comment by christopherlagerhult** [comment]
review @emilwestergren @erikhagglund

**Comment by marcusnaslund** [comment]
Conflicts. Rebase.

File: source/base/Profiler.ooc [hunk 2]
```diff
@@ -75,3 +75,70 @@ Profiler: class {
 }
 
 GlobalCleanup register(|| Profiler free~all())
+
+version (benchmark) {
+Benchmark: class {
+	delimiter := static '|'
+	writeOutput: static func {
+		if (This _contextsActive()) {
```
Comment (emilwestergren) [inline source/base/Profiler.ooc]: If you hard code stuff like this then it should definitely not be in this repo.

Comment (christopherlagerhult) [inline source/base/Profiler.ooc]: Would you approve of changing it to `allProfilersActive()` instead.

**Comment by christopherlagerhult** [comment]
Updated

File: source/base/Benchmarker.ooc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		this _name = name + this _instanceCounter
+	}
+	registerProfiler: func (profiler: Profiler) {
+		if (profiler != null)
+			this _profilers add(profiler)
+	}
+	log: func {
+		filePath := this _name + ".txt"
+		version (android)
+			filePath = "/data/media/0/DCIM/Camera/" + filePath
```
Comment (marcusnaslund) [inline source/base/Benchmarker.ooc]: use `&` instead of `+` to avoid leaking memory

File: source/base/Benchmarker.ooc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		writer := FileWriter new(filePath)
+		for (i in 0 .. this _profilers count) {
+			output := this _profilers[i] name + ">" + this _profilers[i] averageTime + "|\n"
+			writer write(output)
+		}
+		writer free()
+	}
+	free: override func {
+		this _instanceCounter -= 1
```
Comment (marcusnaslund) [inline source/base/Benchmarker.ooc]: `This`
But what do you need this for?

File: source/base/Benchmarker.ooc [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+import Profiler
+import io/FileWriter
+
+Benchmarker: class {
+	_instanceCounter := static 0
+	_name: String
```
Comment (marcusnaslund) [inline source/base/Benchmarker.ooc]: `_name` is never freed

**Comment by marcusnaslund** [comment]
Please add a simple unit test for the `Benchmarker` class.

File: source/base/Benchmarker.ooc (hunk 4 above)
Comment (christopherlagerhult) [inline source/base/Benchmarker.ooc]: It's used to create unique names for the outputfiles. I might change this for a better solution.

Comment (marcusnaslund) [inline source/base/Benchmarker.ooc]: Ahh I see. It's OK, I was just curious

Comment (marcusnaslund) [inline source/base/Benchmarker.ooc]: Or wait, how do you get a unique name if you are also decreasing this counter?

Comment (christopherlagerhult) [inline source/base/Benchmarker.ooc]: Bad choice of words. I need to distinguish between all simultaneous contexts so they don't overwrite each other. 

Not sure yet if we want to generate a NEW file for each context.

File: source/base/Benchmarker.ooc [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+import Profiler
+import io/FileWriter
+
+Benchmarker: class {
+	_instanceCounter := static 0
```
Comment (emilwestergren) [inline source/base/Benchmarker.ooc]: This shouldn't be needed. It's up to the user of this class to supply a unique name.

File: test/base/BenchmarkerTest.ooc [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+use base
+use unit
+import io/[File, FileWriter]
+
+BenchmarkerTest: class extends Fixture {
+	outputDir := "test/base/output"
+	outputFile := "#{this outputDir}/benchmarker.txt"
```
Comment (marcusnaslund) [inline test/base/BenchmarkerTest.ooc]: You can't do this. `outputFile := "test/base/output/benchmarker.txt"`

Comment (marcusnaslund) [inline test/base/BenchmarkerTest.ooc]: @christopherlagerhult

**Comment by marcusnaslund** [comment]
Messed up, so I have to merge this now. If you have any more comments please make them @emilwestergren and then @christopherlagerhult can make a new PR later

**Pull request merged**